---
title: "流感疫苗课题：TIV和宿主菌群的相互影响分析"
output: 
html_document:
      toc: yes
      theme: cerulean
---

          
```{r setup, include=FALSE}
	knitr::opts_chunk$set(echo = TRUE, warning = FALSE, tidy=TRUE, collapse=TRUE)
```

```{r}
# 人员：李辉
# 导师：丁涛 教授
# 概要：纳入167名接种流感疫苗（TIV）的中学生，采集接种疫苗前后的咽拭子、粪便和血液样本，16S测序咽部和肠道菌群，2018级部分学生stage1-3的肠道和咽部菌群宏基因组测序，ELISA检测TIV疫苗应答、LPS和Flagelin等指标，FACS检查非特异性免疫细胞亚群。
#以研究：
#a. 宿主菌群对流感疫苗免疫应答强度的影响；
#b. 接种流感疫苗对上呼吸道和肠道菌群的影响。
```

# 1.流感疫苗接种的免疫应答及影响接种效果的宿主因素

## 1.1 接种疫苗后的免疫应答情况；接种史影响疫苗应答

```{r}
library(ggplot2)
library(plyr)
library(readr)
library(reshape2)
library(patchwork) #拼图
library(cowplot)
library(ggpubr) #统计
	
#载入疫苗接种和基本信息数据
#仅选择2017和2018级数据，以简化分组：有-无接种史学生
	
#数据和图表
##meta_immune_data.csv"
##图p2017_1: 有疫苗接种史学生的滴度分布概率直方图
##图p2018_1: 无接种史学生的滴度分布概率直方图
##图p2017_2: 有疫苗接种史学生的接种疫苗前后变化哑铃图
##图p2018_2: 无疫苗接种史学生的接种疫苗前后变化哑铃图
	
titer<- read.csv("C:/Users/hui/Desktop/TIV16S2020.lihui/processing.data/meta_immune_data.csv",header = T, sep = ',', row.names = 1)
	
col<- c("#00AFBB", "#FC4E07","#36BED9")
#选择接种前、接种后和年级row
sub_titer<- titer[,c("pre_H1N1","post_H1N1","Grade")]
data_2<- melt(sub_titer)
#去除有缺失值的col
data_2<- na.omit(data_2)
#分别提取2017和2018级的数据子集
data_2016 <- subset(data_2, Grade %in% c("G_2016"))
data_2017 <- subset(data_2, Grade %in% c("G_2017"))
data_2018 <- subset(data_2, Grade %in% c("G_2018"))

#柱状概率分布图的可视化，2016级
cdat_2016 <- ddply(data_2016, "variable", summarise, Titer.mean=mean(value))
p2016_1<- ggplot(data_2016, aes(x=value, fill=variable)) +
	    geom_histogram(binwidth=1, alpha=.5, position="identity")+
	    scale_fill_manual(values=col)+
	    geom_vline(data=cdat_2016, aes(xintercept=Titer.mean,  colour=c("#FC4E07","#00AFBB" )), linetype="dashed", size=1) +
	    scale_x_continuous(limits=c(-1,10), breaks=seq(0,10,1))+
	    scale_y_continuous(limits=c(0,20), breaks=seq(0,20,5))+
	    theme_bw()+
	    theme(axis.text.x = element_text(size=12) ,axis.title.x=element_blank(),axis.title.y=element_blank()) + 
	    theme(legend.position="none")+
	    coord_flip()
	
#柱状概率分布图的可视化，2017级
cdat_2017 <- ddply(data_2017, "variable", summarise, Titer.mean=mean(value))
p2017_1<- ggplot(data_2017, aes(x=value, fill=variable)) +
	    geom_histogram(binwidth=1, alpha=.5, position="identity")+
	    scale_fill_manual(values=col)+
	    geom_vline(data=cdat_2017, aes(xintercept=Titer.mean,  colour=c("#FC4E07","#00AFBB" )), linetype="dashed", size=1) +
	    scale_x_continuous(limits=c(-1,10), breaks=seq(0,10,1))+
	    scale_y_continuous(limits=c(0,20), breaks=seq(0,20,5))+
	    theme_bw()+
	    theme(axis.text.x = element_text(size=12) ,axis.title.x=element_blank(),axis.title.y=element_blank()) + 
	    theme(legend.position="none")+
	    coord_flip()
	    
#柱状概率分布图的可视化，2018级
cdat_2018 <- ddply(data_2018, "variable", summarise, Titer.mean=mean(value))
p2018_1<- ggplot(data_2018, aes(x=value, fill=variable)) +
	    geom_histogram(binwidth=1, alpha=.5, position="identity") +
	    geom_vline(data=cdat_2018, aes(xintercept=Titer.mean,  colour=c("#FC4E07","#00AFBB" )), linetype="dashed", size=1)+
	    scale_fill_manual(values=col) +
	    scale_x_continuous(limits=c(-1,10), breaks=seq(0,10,1)) +
	    scale_y_continuous(limits=c(0,20), breaks=seq(0,20,5))+
	    theme_bw()+
	    theme(axis.text.x = element_text(size=12),axis.title.x=element_blank(),axis.title.y=element_blank()) + 
	    theme(legend.position="none")+
	    theme(title=element_text(color="#4F4F4F"))+
	    coord_flip()
	
#重新导入数据，不设置rowname
titer2<- read.csv("C:/Users/hui/Desktop/TIV16S2020.lihui/processing.data/meta_immune_data.csv",header = T, sep = ',')

titer2016 <- subset(titer2, Grade %in% c("G_2016"))
sub_titer1<- titer2016[c("id","pre_H1N1", "post_H1N1")]
sub_titer1$id <- factor(sub_titer1$id, levels = sub_titer1$id[order(sub_titer1$pre_H1N1,sub_titer1$post_H1N1)])
mydata<-melt(sub_titer1,id.vars='id')
	
#接种疫苗前后变化哑铃图，2016级 
p2016_2<- ggplot(mydata, aes(value,id,fill=variable)) +
	    geom_line(aes(group = id), size = 1, color="grey") +
	    geom_point(shape=21,size=3,colour="black")+
	    scale_fill_manual(values=col)+
	    theme(
	        axis.title=element_text(size=13,face="plain",color="white"),
	        legend.title=element_text(size=12,face="plain",color="black"),
	        legend.background = element_blank(),
	        legend.position = c(0.85,0.12)
	    )+ 
	    theme(axis.text.x = element_text(angle=90, hjust=1, vjust=.5))+
	    scale_x_continuous(limits=c(-1, 10), breaks=seq(0, 10, 1))+ 
	    coord_flip()+ 
	    background_grid(major = "xy", minor = "none")+ 
	    theme(legend.position="none")+
	    theme(axis.text.y = element_text(size=12))


titer2017 <- subset(titer2, Grade %in% c("G_2017"))
sub_titer1<- titer2017[c("id","pre_H1N1", "post_H1N1")]
sub_titer1$id <- factor(sub_titer1$id, levels = sub_titer1$id[order(sub_titer1$pre_H1N1,sub_titer1$post_H1N1)])
mydata<-melt(sub_titer1,id.vars='id')
	
#接种疫苗前后变化哑铃图，2017级 
p2017_2<- ggplot(mydata, aes(value,id,fill=variable)) +
	    geom_line(aes(group = id), size = 1, color="grey") +
	    geom_point(shape=21,size=3,colour="black")+
	    scale_fill_manual(values=col)+
	    theme(
	        axis.title=element_text(size=13,face="plain",color="white"),
	        legend.title=element_text(size=12,face="plain",color="black"),
	        legend.background = element_blank(),
	        legend.position = c(0.85,0.12)
	    )+ 
	    theme(axis.text.x = element_text(angle=90, hjust=1, vjust=.5))+
	    scale_x_continuous(limits=c(-1, 10), breaks=seq(0, 10, 1))+ 
	    coord_flip()+ 
	    background_grid(major = "xy", minor = "none")+ 
	    theme(legend.position="none")+
	    theme(axis.text.y = element_text(size=12))
	
	
titer2018 <- subset(titer2, Grade %in% c("G_2018"))
sub_titer1<- titer2018[c("id","pre_H1N1", "post_H1N1")]
sub_titer1$id <- factor(sub_titer1$id, levels = sub_titer1$id[order(sub_titer1$pre_H1N1,sub_titer1$post_H1N1)])
mydata<-melt(sub_titer1,id.vars='id')
	
#接种疫苗前后变化哑铃图，2018级
p2018_2<- ggplot(mydata, aes(value,id,fill=variable)) +
	    geom_line(aes(group = id), size = 1, color="black") +
	    geom_point(shape=21,size=3,colour="black")+
	    scale_fill_manual(values=col)+
	    theme(
	        axis.title=element_text(size=13,face="plain",color="white"),
	        legend.title=element_text(size=12,face="plain",color="black"),
	        legend.background = element_blank(),
	        legend.position = c(0.85,0.12)
	    )+ 
	    theme(axis.text.x = element_text(angle=90, hjust=1, vjust=.5))+
	    scale_x_continuous(limits=c(-1, 10), breaks=seq(0, 10, 1))+ 
	    coord_flip()+ 
	    theme(panel.background = element_rect(fill = "#FDF3EC"))+
	    background_grid(major = "xy", minor = "none")+ 
	    theme(axis.text.y = element_text(size=12))+ 
	    theme(legend.title=element_blank(), legend.text = element_text(size = 12))
	

(p2016_2+ p2016_1 + plot_layout(nrow = 1, widths = c(3, 1))) / (p2017_2+ p2017_1 + plot_layout(nrow = 1, widths = c(3, 1))) / (p2018_2 + p2018_1 + plot_layout(nrow = 1, widths = c(3, 1))) #fig001


# 比较有-无疫苗接种记录人群的TIV应答差异

titer$records <- titer$Grade
titer$records[titer$Grade %in% c("G_2016", "G_2017")] = "Record"
titer$records[titer$Grade == "G_2018"] = "no Record"

titer$records <- factor(titer$records, levels = c( "Record","no Record"), ordered=TRUE)

my_comparisons <- list(c("Record", "no Record"))

p.fig1b<- ggplot(titer, aes(x=records, y=diff_H1N1), shape=8 ) +
	    geom_boxplot(aes(x=records, fill=records), notch=FALSE) +
	    stat_summary(aes(x=records), fun.y=mean, geom="point") +
	    geom_jitter(width=0.05, height=0, size=2) +
	    scale_fill_manual(values=c("#b34e7e", "#1b7837"))+
	    theme(axis.title.x =element_blank())+
      theme_bw()+
	    theme(axis.text.x = element_text(size=14),axis.text.y = element_text(size=14)) + 
	    theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank()) + 
	    theme(legend.title = element_blank())+ 
	    theme(legend.position='none')+
	    theme(title=element_text(size=18,color="#4F4F4F"))+
	    scale_y_continuous(limits=c(0, 11), breaks=seq(0, 10, 2))+
	    theme(plot.title = element_text(hjust = 0.5))+
	    stat_compare_means(comparisons = my_comparisons, paired = FALSE, label = "p.signif", color="black", size=6)+
	    labs(y="H1N1 titer change (log)", x="")

p.fig1b #fig002
```

## 1.2 本底抗体水平与滴度变化有显著关联

```{r}
# 使用spearman关联分析本底滴度对疫苗应答的影响
# 选择无疫苗接种记录（史）的2018级数据
	   
titer<- read.csv("C:/Users/hui/Desktop/TIV16S2020.lihui/processing.data/meta_immune_data.csv",header = T, sep = ',',row.names = 1)
sub_titer<- titer[,c("pre_H1N1","post_H1N1", "diff_H1N1", "Grade", "h1n1_pre_group")]
data_3<- na.omit(sub_titer)
sub_data <- subset(data_3, Grade %in% c("G_2018"))

 p.pre.cahnge<- ggplot(data=sub_data, aes(x = pre_H1N1, y = diff_H1N1)) +
	        geom_point(alpha=0.6, color="black",size=2)+ 
	        geom_smooth(method=lm, size=2, color="black")+ 
	        labs(y= "H1N1 titer change(log)", x="pre_H1N1 titer(log)")+
	        theme(axis.text.x = element_text(size=12),axis.text.y = element_text(size=12)) +
	        theme(legend.title = element_blank())+ 
	        scale_x_continuous(limits=c(0, 8), breaks=seq(0, 8, 1))+
	        scale_y_continuous(limits=c(0, 10), breaks=seq(0, 10, 1))+
	        theme(title=element_text(size=14,color="#4F4F4F"))+
	        theme(legend.position="none")+
          theme_bw()+
	        theme(axis.text.x = element_text(size=14),axis.text.y = element_text(size=14)) + 
	        theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank()) + 
	        theme(legend.title = element_blank())+ 
	        stat_cor(data=sub_data, method = "pearson", color="black")

 p.pre.cahnge #fig003
 
# 比较接种前本底抗体阴性和阳性人群的疫苗应答差异
my_comparisons <- list(c("Negative", "Positive"))
p.fig1c<- ggplot(sub_data, aes(x=h1n1_pre_group, y=diff_H1N1), shape=8 ) +
	    geom_boxplot(aes(x=h1n1_pre_group, fill=h1n1_pre_group), notch=FALSE) +
	    stat_summary(aes(x=h1n1_pre_group), fun.y=mean, geom="point") +
	    geom_jitter(width=0.05, height=0, size=2) +
	    scale_fill_manual(values=c("#533633", "#149684"))+
	    theme(axis.title.x =element_blank())+
      theme_bw()+
	    theme(axis.text.x = element_text(size=14),axis.text.y = element_text(size=14)) + 
	    theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank()) + 
	    theme(legend.title = element_blank())+ 
	    theme(legend.position='none')+
	    theme(title=element_text(size=18,color="#4F4F4F"))+
	    scale_y_continuous(limits=c(0, 11), breaks=seq(0, 10, 2))+
	    theme(plot.title = element_text(hjust = 0.5))+
	    stat_compare_means(comparisons = my_comparisons, paired = FALSE, label = "p.signif", color="black", size=6)+
	    labs(y="", x="")
	    
p.fig1c #fig004
	    
``` 

## 1.3 BMI和性别对疫苗接种效果无显著影响

```{r}
# 选择无疫苗接种史的2018级且接种前本低抗体阴性学生
#BMI对滴度变化的影响
titer<- read.csv("C:/Users/hui/Desktop/TIV16S2020.lihui/processing.data/meta_immune_data.csv",header = T, sep = ',',row.names = 1)
sub_titer1<- titer[,c("diff_H1N1", "Grade", "h1n1_pre_group", "BMI", "Gender")]
sub_titer2<- subset(sub_titer1, Grade == "G_2018" & h1n1_pre_group == "Negative")

p.cor.BMI<- ggplot(data=sub_titer2, aes(x = BMI, y = diff_H1N1)) +
	        geom_point(alpha=0.6, color="black",size=2)+ 
	        geom_smooth(method=lm, size=2, color="black")+ 
	        labs(y= "H1N1 titer change(log)", x="BMI")+
          theme_bw()+
	        theme(axis.text.x = element_text(size=14), axis.text.y = element_text(size=14)) +
	        theme(legend.title = element_blank())+ 
	        scale_y_continuous(limits=c(0, 10), breaks=seq(0, 10, 1))+
	        theme(title=element_text(size=14,color="#4F4F4F"))+
	        theme(legend.position="none")+
	        stat_cor(data=sub_titer2, method = "spearman", color="black", size=6)

# 性别对滴度变化的影响
my_comparisons <- list(c("Female", "Male"))
p.n_p.gender<-  ggplot(sub_titer2, aes(x=Gender,  y=diff_H1N1), shape=8 ) +
    geom_boxplot(aes(x=Gender, fill=Gender), notch=FALSE) +
    stat_summary(aes(x=Gender), fun.y=mean, geom="point") +
    geom_jitter(width=0.2, size=2) +
        scale_fill_manual(values=c( "#F8766D","#7F7F7F"))+
    theme(axis.title.x =element_blank())+
    theme_bw()+ 
    theme(axis.text.x = element_text(size=14),axis.text.y = element_text(size=14)) + 
    theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank()) + 
    theme(legend.title = element_blank())+ 
    theme(legend.position='none')+
    theme(title=element_text(size=14,color="#4F4F4F"))+
    scale_y_continuous(limits=c(0, 11), breaks=seq(0, 10, 2))+
    theme(plot.title = element_text(hjust = 0.5))+
    stat_compare_means(comparisons = my_comparisons, paired = FALSE, label = "p.signif", color="black", size=6)+
    labs(y="")

p.cor.BMI | p.n_p.gender #fig005
```

## 1.4 宿主接种前的非特异性免疫细胞亚群与滴度变化无显著关联

```{r}
library(corrplot)

#选择无疫苗接种史的2018级且接种前本低抗体阴性学生

titer<- read.csv("C:/Users/hui/Desktop/TIV16S2020.lihui/processing.data/meta_immune_data.csv",header = T, sep = ',',row.names = 1)
sub_titer<- subset(titer, Grade == "G_2018" & h1n1_pre_group == "Negative")

#T细胞、CD4 T及其记忆细胞亚群
#spearman相关分析

data_4t<- sub_titer[,c("diff_H1N1", "pre_T", "pre_4T", "pre_4NaiveT", "pre_4TCM", "pre_4TEM")]
data2_4t2<- melt(data.frame(data_4t), id="diff_H1N1")
col2<- c("#000000", "#CC0033",  "#003399", "#FF6600", "#1F6F43")
p4t<- ggplot(data2_4t2, aes(x = diff_H1N1, y = value, group= variable, colour=variable)) +
    geom_point(alpha=0.3)+
    scale_colour_manual(values=col2) +
    geom_smooth(size=1, method =lm )+ 
    labs(y= "Count %", x="Titer change")+
    scale_x_continuous(limits=c(0, 7), breaks=seq(0, 7, 1))+
    theme(axis.text.x =element_text(size=12), axis.text.y = element_text(size=12)) +
    theme(legend.title = element_blank())+ 
    theme(title=element_text(size=14,color="#4F4F4F"))+
    theme(legend.position="top")+
    stat_cor(method = "spearman")
p4t

#CD8 T及其记忆细胞亚群
#spearman相关分析

data_8t<- sub_titer[,c("diff_H1N1", "pre_8TEMRA", "pre_8T", "pre_8NaiveT", "pre_8TCM", "pre_8TEM")]
data2_8t2<- melt(data.frame(data_8t), id="diff_H1N1")
col2<- c("#000000", "#CC0033",  "#003399", "#FF6600", "#1F6F43")
p8t<- ggplot(data2_8t2, aes(x = diff_H1N1, y = value, group= variable, colour=variable)) +
    geom_point(alpha=0.3)+
    scale_colour_manual(values=col2) +
    geom_smooth(size=1, method = lm, linetype = "dashed")+ 
    scale_x_continuous(limits=c(0, 7), breaks=seq(0, 7, 1))+
    labs(y= "Count %", x="Titer change")+
    theme(axis.text.x =element_text(size=12), axis.text.y = element_text(size=12)) +
    theme(legend.title = element_blank())+ 
    theme(title=element_text(size=14,color="#4F4F4F"))+
    theme(legend.position="top")+
    stat_cor(method = "spearman")
p8t

#B记忆细胞亚群
#spearman相关分析

data_b<- sub_titer[, c("diff_H1N1", "pre_MsB", "pre_MuB", "pre_NmB", "pre_ImB")]
data2_b2<- melt(data.frame(data_b), id="diff_H1N1")
pb<- ggplot(data2_b2, aes(x = diff_H1N1, y = value, group= variable, colour=variable)) +
    geom_point(alpha=0.3)+
    scale_colour_manual(values=col2) + 
    geom_smooth(size=1, method = lm, linetype = "dotdash")+ 
    scale_x_continuous(limits=c(0, 7), breaks=seq(0, 7, 1))+
    labs(y= "Count %", x="Titer change")+
    theme(axis.text.x = element_text(size=12), axis.text.y = element_text(size=12)) +
    theme(legend.title = element_blank())+ 
    theme(title=element_text(size=14,color="#4F4F4F"))+
    theme(legend.position="top")+
    stat_cor(method = "spearman")
pb

#fig006
```


## 1.5 TIV强弱应答组的划分

```{r}

sub_titer<- subset(titer, Grade=="G_2018"&h1n1_pre_group=="Negative")
sub_titer1<- sub_titer
sub_titer1$group<- sub_titer1$diff_H1N1
sub_titer1$group[sub_titer1$group >5] = "high"
sub_titer1$group[sub_titer1$group <6] = "low"
p1<- ggplot(sub_titer1, aes(x=Grade,  y=diff_H1N1), shape=8 ) +
    geom_jitter(aes(color=group), width=0.2, height=0, size=3) +
    scale_color_manual(values=c("#E41A1C", "#377EB8"))+
    geom_hline(yintercept = 5.5, color = 'gray', size = 0.4)+ 
    theme(axis.title.x =element_blank())+
    theme_bw()+ 
    theme(axis.text.x = element_text(size=14),axis.text.y = element_text(size=14)) + 
    theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank()) + 
    theme(legend.title = element_blank())+ 
    theme(legend.position='none')+
    theme(title=element_text(size=14,color="#4F4F4F"))+
    scale_y_continuous(limits=c(0, 10), breaks=seq(0, 10, 1))+
    theme(plot.title = element_text(hjust = 0.5))+
    theme(axis.title.x =element_blank(), axis.title.y =element_blank())+
    scale_x_discrete(breaks=c("G_2018"), labels=c("no Record & Negative"))

p1 #fig007

sub_titer2<- sub_titer1[, c(20:24,71)]
sub_titer3<- melt(data.frame(sub_titer2), id="group")
my_comparisons <- list(c("low", "high"))
p2<- ggplot(sub_titer3, aes(x=group,  y=value), shape=8 ) +
    geom_boxplot(aes(x=group, fill=group), notch=FALSE) +
    stat_summary(aes(x=group), fun.y=mean, geom="point") +
    geom_jitter(width=0.2, size=2) +
    scale_fill_manual(values=c("#E41A1C", "#377EB8"))+
    theme(axis.title.x =element_blank())+
    theme_bw()+ 
    theme(axis.text.x = element_text(size=14),axis.text.y = element_text(size=14)) + 
    theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank()) + 
    theme(legend.title = element_blank())+ 
    theme(legend.position='none')+
    theme(title=element_text(size=14,color="#4F4F4F"))+
    theme(plot.title = element_text(hjust = 0.5))+
    stat_compare_means(comparisons = my_comparisons, paired = FALSE,color="red")+
    theme(axis.title.x =element_blank(), axis.title.y =element_blank())+
    coord_flip()+
    facet_wrap(.~variable,scales="free", ncol=1)+
    theme(strip.text = element_text(colour = 'black', face = 'bold', size = rel(1.2)), strip.background = element_rect(fill = 'white', colour = 'black', size = rel(2), linetype = 1))

p2 #fig008
```

## 1.6 强-弱组肠道菌群结构有显著差异，呼吸道菌群结构差异不显著

```{r}
#使用phyloseq对象输出的Rdata
# 选取无疫苗接种史且本底抗体阴性的学生
# 滴度变化大于等于6的划分为高反应组，小于等于5的划分为低反应组
#肠道菌群vs滴度变化

library(microbiome)
library(phyloseq)
library(vegan)

load('C:/Users/hui/Desktop/TIV16S2020.lihui/processing.data/physeq.tiv16sall.Rdata')
physeq<- physeq.tiv16sall
titer<- read.csv("C:/Users/hui/Desktop/TIV16S2020.lihui/processing.data/meta_immune_data.csv",header = T, sep = ',',row.names = 1)
sub_titer<- subset(titer, Grade=="G_2018"&h1n1_pre_group=="Negative")
sub_titer1<- sub_titer[, c("pre_H1N1", "diff_H1N1")]
sub_titer1$group<- sub_titer1$diff_H1N1
sub_titer1$group[sub_titer1$group >5] = "high"
sub_titer1$group[sub_titer1$group <6] = "low"
student_id<- rownames(sub_titer1)
physeq.tib3<- subset_samples(physeq, type=="fecal"&national=="tibetan"& grade=="grade_18"&stage=="stage1")
physeq.tib<-  physeq.tib3
physeq.tib.p<-  prune_taxa(taxa_sums(physeq.tib) > 0, physeq.tib)
sample_data(physeq.tib.p)$human <- get_variable(physeq.tib.p, "student_id") %in% student_id
sub_physeq.tib.p<- subset_samples(physeq.tib.p, human=="TRUE")
physeq.fp<-  prune_taxa(taxa_sums(sub_physeq.tib.p) > 0, sub_physeq.tib.p)
physeq.fp1<- filter_taxa(physeq.fp, function(x) sum(x > 3) > 1, TRUE)

#基于bray距离的adonis组间差异性检验
dis_bray.f<- phyloseq::distance(physeq.fp1, "bray")
set.seed(002)
adon.results<-adonis(dis_bray.f~ sub_titer1$group, perm=999)
knitr::kable(adon.results[["aov.tab"]])

#合并OTU至属水平，结果输出至工作目录
#输出文件更改名称为：genus.fp.forcore.txt
#library(yingtools2)
#lefse(physeq.fp, class="student_id", levels = "Genus")

#选择在95%样本中丰度不小于千分之一的物种
#去除无注释信息的物种
genus <- read.table('C:/Users/hui/Desktop/TIV16S2020.lihui/output.data/genus.fp.forcore.txt', header = T, sep = '\t', row.names = 1)
core.taxa.f <- core_members(genus, detection = 0.001, prevalence = 0.95)
genus.core<- as.data.frame(t(genus[core.taxa.f[1:17],]))

#基于OTU的NMDS排序
#将得到的17个核心属与β多样性分布进行线性拟合
library(vegan)

otu<- t(physeq.fp1@otu_table@.Data)
nmds <- metaMDS(otu, distance = 'bray', k = 2)

#创建排序结果分组信息的数据集
data.all<- cbind(nmds[["points"]], genus.core, sub_titer1)
nmds_plot.fecal <- ggscatter(data.all, x= "MDS1", y = "MDS2", 
                             color = "group", 
                             size = 3,
                             palette = "Set1", 
                             ellipse = TRUE, 
                             mean.point = FALSE, 
                             star.plot = TRUE, 
                             ellipse.level = FALSE,  
                             ggtheme = theme_minimal()) +
    labs(x = "NMDS1", y = "NMDS2") +
    ggtitle("NMDS")+
    theme(panel.grid = element_line(color = 'gray', linetype = 2, size = 0.1), panel.background = element_rect(color = 'black', fill = 'transparent'), legend.key = element_rect(fill = 'transparent')) + #去掉背景框
    geom_vline(xintercept = 0, color = 'gray', size = 0.4) + 
    geom_hline(yintercept = 0, color = 'gray', size = 0.4) + 
    theme(axis.text.x = element_text(size=16),axis.text.y = element_text(size=16)) + 
    labs(title="Bray_Curtis distance of Fecal samples")+
    theme(title=element_text(size=16,color="#4F4F4F"))+
    theme(plot.title = element_text(hjust = 0.5))+ 
    theme(legend.title = element_blank(),legend.text = element_text(colour="#4F4F4F", size = 14))+
    annotate("text",x=-0.35,y=0.4,parse=TRUE,size=4,label=paste('Adonis-P:',adon.results$aov.tab$`Pr(>F)`[1]),family="serif",fontface="italic",colour="black")+
    annotate("text",x=-0.35,y=0.35,parse=TRUE,size=4,label=paste("Stess: ", nmds[["stress"]]),family="serif",fontface="italic",colour="black")+
    theme(legend.position="none")

nmds_plot.fecal #fig009

data.all <-  read.csv("C:/Users/hui/Desktop/fig1g.csv",header = T, sep = ',',row.names = 1)

my_comparisons <- list(c("low", "high"))
p.Prevotella<- ggplot(data.all, aes(x=group, y=g_Prevotella*100), shape=8 ) +
    theme_bw()+ 
    geom_boxplot(aes(x=group, fill=group), notch=FALSE, outlier.colour="grey") +
    stat_summary(aes(x=group), fun.y=mean, geom="point", color="orange") +
    geom_jitter(width=0.2, size=3) +
    scale_fill_manual(values=c("#E41A1C", "#377EB8"))+
    theme(axis.title.x =element_blank())+ #设置x.y标题上文本的名称
    theme(axis.text.x = element_text(size=14),axis.text.y = element_text(size=14)) + # 设置x.y坐标上文本大小
    theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank()) + #不显示网格线
    theme(legend.title = element_blank())+ # 不显示图例的标题
    theme(legend.position='none')+
    theme(title=element_text(size=18,color="#4F4F4F"))+
    theme(plot.title = element_text(hjust = 0.5))+
    labs(y="Relative Aboundance %")+
    stat_compare_means(comparisons = my_comparisons, paired = FALSE,  color="red")+
    labs(title="Prevotella")

p.Prevotella #fig010

my_comparisons <- list(c("low", "high"))
pcopri<- ggplot(data.all, aes(x=group, y=Prevotella_copri*100), shape=8 ) +
    theme_bw()+ 
    geom_boxplot(aes(x=group, fill=group), notch=FALSE, outlier.colour="grey") +
    stat_summary(aes(x=group), fun.y=mean, geom="point", color="orange") +
    geom_jitter(width=0.2, size=3) +
    scale_fill_manual(values=c("#E41A1C", "#377EB8"))+
    theme(axis.title.x =element_blank())+ #设置x.y标题上文本的名称
    theme(axis.text.x = element_text(size=14),axis.text.y = element_text(size=14)) + # 设置x.y坐标上文本大小
    theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank()) + #不显示网格线
    theme(legend.title = element_blank())+ # 不显示图例的标题
    theme(legend.position='none')+
    theme(title=element_text(size=18,color="#4F4F4F"))+
    theme(plot.title = element_text(hjust = 0.5))+
    labs(y="Relative Aboundance %")+
    stat_compare_means(comparisons = my_comparisons, paired = FALSE,  color="red")+
    labs(title="Prevotella copri")

p.copri


# MDI
add_mdi<-function (ps) 
{
    mdi_dec <- c("g_Streptococcus", "g_Prevotella")
    mdi_down <- rownames(tax_table(ps)[which(tax_table(ps)[, "Genus"] %in% mdi_dec), ])
    mdi_down_ps <- prune_taxa(mdi_down, ps)
    mdi_inc <- c("g_Collinsella", "g_Hydrogenoanaerobacterium", "g_Enterococcus")
    mdi_up <- rownames(tax_table(ps)[which(tax_table(ps)[, "Genus"] %in% 
                                               mdi_inc), ])
    mdi_up_ps <- prune_taxa(mdi_up, ps)
    if (!is.null(ps@sam_data$MDI)) {
        message("MDI already exists in sample_data() and will be overwritten.")
    }
    ps@sam_data$MDI <- log(sample_sums(mdi_up_ps) + 1/sample_sums(mdi_down_ps) + 
                               1)
    ps@sam_data$MDI[is.infinite(ps@sam_data$MDI)] <- NA
    if (sum(is.na(ps@sam_data$MDI)) > 0) {
        message(paste0("There were ", sum(is.na(ps@sam_data$MDI)), 
                       " samples (out of ", nsamples(ps), " total) where MDI = NA."))
    }
    return(ps)
}

test1 <- add_mdi(ps)
data.all<- cbind(meta(test1), sub_titer1)
my_comparisons <- list(c("low", "high"))
ggplot(data.all, aes(x=group, y=MDI), shape=8 ) +
    theme_bw()+ 
    geom_boxplot(aes(x=group, fill=group), notch=FALSE, outlier.colour="grey") +
    stat_summary(aes(x=group), fun.y=mean, geom="point", color="orange") +
    geom_jitter(width=0.2, size=3) +
    scale_fill_manual(values=c("#E41A1C", "#377EB8"))+
    theme(axis.title.x =element_blank())+ #设置x.y标题上文本的名称
    theme(axis.text.x = element_text(size=14),axis.text.y = element_text(size=14)) + # 设置x.y坐标上文本大小
    theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank()) + #不显示网格线
    theme(legend.title = element_blank())+ # 不显示图例的标题
    theme(legend.position='none')+
    theme(title=element_text(size=18,color="#4F4F4F"))+
    theme(plot.title = element_text(hjust = 0.5))+
    stat_compare_means(comparisons = my_comparisons, paired = FALSE,  color="black")

# nmds拟合物种变量
nmds_ef <- envfit(nmds~., data = genus.core, perm = 999, choices = c(1,2), display = 'sites')
nmds_ef$vectors$pvals <- p.adjust(nmds_ef$vectors$pvals, method = 'bonferroni' ) 
tbnmds_ef <- data.frame(cbind(nmds_ef$vectors$arrows, nmds_ef$vectors$r, nmds_ef$vectors$pvals))
names(tbnmds_ef) <- c('PC1', 'PC2', 'r2', 'p.adj')
nmds_ef_sign <- tbnmds_ef[which(tbnmds_ef$p.adj < 0.05),1:2]
nmds_ef_sign$genus.core.sel<- rownames(nmds_ef_sign)

data.all<- cbind(nmds[["points"]], genus.core, sub_titer1)
nmds_plot <- ggplot(data.all, aes(MDS1, MDS2)) +
    theme(panel.grid = element_line(color = 'gray', linetype = 2, size = 0.1), panel.background = element_rect(color = 'black', fill = 'transparent'), legend.key = element_rect(fill = 'transparent')) + #去掉背景框
    geom_vline(xintercept = 0, color = 'gray', size = 0.4) + 
    geom_hline(yintercept = 0, color = 'gray', size = 0.4) + 
    geom_point(aes(color = group), size = 3) + #可在这里修改点的透明度、大小
    scale_color_manual(values=c("red", "blue")) + #可在这里修改点的颜色
    guides(fill = guide_legend(order = 1), shape = guide_legend(order = 2), color = guide_legend(order = 3))+  #设置图例展示顺序
    labs(x = "NMDS1", y = "NMDS2")+
    theme(axis.text.x = element_text(size=18),axis.text.y = element_text(size=18)) + # 设置x.y坐标上文本大小
    labs(title="Bray_Curtis distance of fecal samples")+
    theme(title=element_text(size=18,color="#4F4F4F"))+
    theme(plot.title = element_text(hjust = 0.5))+ 
    theme(legend.title = element_blank(),legend.text = element_text(colour="#4F4F4F", size = 16))+
    geom_segment(data = nmds_ef_sign, aes(x = 0,y = 0, xend = PC1/2, yend = PC2/2), arrow = arrow(length = unit(0.2, 'cm')), size = 1, color = 'black') +
    geom_text(data = nmds_ef_sign, aes(PC1/2 * 0.9, PC2/2 * 0.9, label = genus.core.sel), color = 'black', size = 5)+
    theme(legend.position="none")

nmds_plot #fig011

#咽部菌群vs滴度变化
sub_titer<- subset(titer, Grade=="G_2018"&h1n1_pre_group=="Negative")
sub_titer1<- sub_titer[, c("pre_H1N1", "diff_H1N1")]
sub_titer1$group<- sub_titer1$diff_H1N1
sub_titer1$group[sub_titer1$group >5] = "high"
sub_titer1$group[sub_titer1$group <6] = "low"
student_id<- rownames(sub_titer1)
physeq.tib3<- subset_samples(physeq, type=="oroph"&national=="tibetan"& grade=="grade_18"&stage=="stage1")
physeq.tib<-  physeq.tib3
physeq.tib.p<-  prune_taxa(taxa_sums(physeq.tib) > 0, physeq.tib)
sample_data(physeq.tib.p)$human <- get_variable(physeq.tib.p, "student_id") %in% student_id
sub_physeq.tib.p<- subset_samples(physeq.tib.p, human=="TRUE")
physeq.op<-  prune_taxa(taxa_sums(sub_physeq.tib.p) > 0, sub_physeq.tib.p)
physeq.op1<- filter_taxa(physeq.op, function(x) sum(x > 3) > 1, TRUE)

#基于bray距离的adonis组间差异性检验
dis_bray.o<- phyloseq::distance(physeq.op1, "bray")
adon.results<-adonis(dis_bray.o ~ sub_titer1$group, perm=999)
knitr::kable(adon.results[["aov.tab"]])

#合并OTU至属水平，结果输出至工作目录
#输出文件更改名称为：genus.op.forcore.txt
#library(yingtools2)
#lefse(physeq.op, class="student_id", levels = "Genus")

#选择在95%样本中丰度不小于千分之一的物种
#去除无注释信息的物种
genus <- read.table('C:/Users/hui/Desktop/TIV16S2020.lihui/output.data/genus.op.forcore.txt', header = T, sep = '\t', row.names = 1)
core.taxa.f <- core_members(genus, detection = 0.001, prevalence = 0.95)
genus.core<- as.data.frame(t(genus[core.taxa.f[c(1:16,18)],]))

otu<- t(physeq.op1@otu_table@.Data)
nmds <- metaMDS(otu, distance = 'bray', k = 2)

data.all<- cbind(nmds[["points"]],  sub_titer1)
nmds_plot.oroph <- ggscatter(data.all, x= "MDS1", y = "MDS2", 
          color = "group", 
          size = 3,
          palette = "Set1", 
          ellipse = TRUE, 
          mean.point = FALSE, 
          star.plot = TRUE, 
          ellipse.level = FALSE,  
          ggtheme = theme_minimal()) +
    labs(x = "NMDS1", y = "NMDS2") +
    ggtitle("NMDS")+
    theme(panel.grid = element_line(color = 'gray', linetype = 2, size = 0.1), panel.background = element_rect(color = 'black', fill = 'transparent'), legend.key = element_rect(fill = 'transparent')) + #去掉背景框
    geom_vline(xintercept = 0, color = 'gray', size = 0.4) + 
    geom_hline(yintercept = 0, color = 'gray', size = 0.4) + 
    theme(axis.text.x = element_text(size=18),axis.text.y = element_text(size=18)) + 
    labs(title="Bray_Curtis distance of Oroph samples")+
    theme(title=element_text(size=18,color="#4F4F4F"))+
    theme(plot.title = element_text(hjust = 0.5))+ 
    theme(legend.title = element_blank(),legend.text = element_text(colour="#4F4F4F", size = 16))+
    annotate("text",x=-0.35,y=0.35,parse=TRUE,size=4,label=paste('Adonis-P:',adon.results$aov.tab$`Pr(>F)`[1]),family="serif",fontface="italic",colour="black")+
    annotate("text",x=-0.35,y=0.30,parse=TRUE,size=4,label=paste("Stess: ", nmds[["stress"]]),family="serif",fontface="italic",colour="black")+
    theme(legend.position="none")

nmds_plot.oroph #fig012
```

## 1.7 强-弱组肠道、咽部菌群多样性都无显著差异

```{r, fig.width=10,fig.height=5}



fo.sh <- plot_richness(physeq.op1, "stage",  measures=NULL)
shannon.oroph <- fo.sh$data
shannon.oroph$group <- sub_titer1[as.character(shannon.oroph$student_id),]$group

ff.sh <- plot_richness(physeq.fp1, "stage",  measures=NULL)
shannon.fecal <- ff.sh$data
shannon.fecal$group <- sub_titer1[as.character(shannon.fecal$student_id),]$group
shannon.data <- rbind(shannon.fecal, shannon.oroph)
sub_shannon.data <- subset(shannon.data, variable %in% c("Observed", "Shannon"))

p.alpha<- ggplot(data=shannon.data, aes(x = group, y = value, group=group)) +
    geom_boxplot(aes(x=group, fill=group), notch=FALSE) +
    stat_summary(aes(x=group), fun.y=mean, geom="point") +
    geom_jitter(width=0.2, size=2) +
    scale_fill_manual(values=c("#E41A1C", "#377EB8")) +
    theme(title=element_text(size=14,color="#4F4F4F"))+
    facet_wrap(type~variable, scales = "free", nrow=2)+
    stat_compare_means(comparisons = my_comparisons, paired = FALSE, color="black")+
    theme_bw()+ 
    theme(axis.text.x = element_text(size=14),axis.text.y = element_text(size=14)) + 
    theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank()) + 
    theme(legend.title = element_blank())+ 
    theme(legend.position='none')+
    theme(plot.title = element_text(hjust = 0.5))

p.alpha #fig013

```


# 2.以Prevotella为中心的功能性肠菌集合影响流感疫苗应答强度

## 2.1 高应答人群的肠道菌群更加聚集（提示群体稳定性）

```{r}

titer<- read.csv("C:/Users/hui/Desktop/TIV16S2020.lihui/processing.data/meta_immune_data.csv",header = T, sep = ',',row.names = 1)
sub_titer<- subset(titer, Grade=="G_2018"&h1n1_pre_group=="Negative")
sub_immune <- sub_titer[,c(13,17,21:26,29:31,35,37:40,44:47)]


genus <- read.table('C:/Users/hui/Desktop/meta.genus.txt', header = T, sep = '\t', row.names = 1)
genus1<- t(t(genus)/t(genus)[,1])[-1,rownames(meta(physeq.fp))]


core.taxa.f <- core_members(genus1, detection = 0.001, prevalence = 0.8)
genus.core<- t(genus1[core.taxa.f,])


library(psych)
cortest_psy <- corr.test(sub_titer1, genus.core, method = "spearman")
cortest_psy_sdj <- corr.test(sub_titer1, genus.core, method = "spearman", adjust = "none")
col3 <- colorRampPalette(c("blue", "white", "red"))
corrplot(cortest_psy$r,col=col3(20),method='color',tl.pos='lt', tl.col='black', p.mat=cortest_psy_sdj$p, insig= 'label_sig', sig.level=c(.001, .01, .05), pch.cex= .9, pch.col='black', xpd = T,tl.srt = 45)


library(mixOmics)
species <- read.table('C:/Users/hui/Desktop/meta.species.txt', header = T, sep = '\t', row.names = 1)
species1<- t(t(species)/t(species)[,1])[-1,rownames(meta(physeq.fp))]
core.taxa.f <- core_members(species1, detection = 0.001, prevalence = 0.8)
species.core<- t(species1[core.taxa.f,])

bolite<- read.csv("C:/Users/hui/Desktop/bolite.csv",header = T, sep = ',',row.names = 1)
data.bolite <- subset(bolite, MS2hmdb != "-" | MS2kegg != "-")
metabolite <- t(data.bolite[,c(16:41)])

genefamily <- read.table('C:/Users/hui/Desktop/named_go_genefamilies_unstratified.txt', header = T, sep = '\t', row.names = 1)
gene.data <- t(genefamily[-c(1,2), c(1:26)])
gene <- gene.data[,core_members(t(gene.data), prevalence = 0.8)]


immune<- as.matrix(sub_titer[,c(25:26,29:31,35,37:40,44:47)])
rownames(species.core) <- rownames(immune)
rownames(metabolite) <- rownames(immune)
rownames(gene) <- rownames(immune)


X <- list(species=species.core, metabolite=metabolite, immune=immune, gene=gene)
#Y= as.factor(sub_titer1$group)

list.keepX <- list(species = c(12, 17), metabolite = c(10,5), immune = c(4, 5), gene =c(6,19))

MyResult.diablo <- block.splsda(X, Y, keepX=list.keepX)


plotIndiv(MyResult.diablo, 
          ind.names = FALSE, 
          legend=TRUE, cex=c(1,2),
          title = 'BRCA with DIABLO')

plotVar(MyResult.diablo, var.names = c( FALSE,TRUE, FALSE, FALSE),
        legend=TRUE, pch=c(16,16,16,16), col = c('darkorchid', 'brown1', 'lightgreen', "black"))

plotDiablo(MyResult.diablo, ncomp = 1)

circosPlot(MyResult.diablo, cutoff=0.7)

plotLoadings(MyResult.diablo, comp = 1, contrib = "max")

cimDiablo(MyResult.diablo, color.blocks = c('darkorchid', 'brown1', 'lightgreen', "black"), comp = 1, margin=c(8,20), legend.position = "right")

Myauc.diablo <- auroc(MyResult.diablo, roc.block = "gene", roc.comp = 2)

net<- network(MyResult.diablo, blocks = c(1,2,3,4),
        color.node = c('darkorchid', 'brown1', 'lightgreen', "black"), 
        cutoff = 0.6)

## 分别计算高、低应答人群的Divergence，两个人群有显著差异
titer<- read.csv("C:/Users/hui/Desktop/TIV16S2020.lihui/processing.data/meta_immune_data.csv",header = T, sep = ',',row.names = 1)
sub_titer<- subset(titer, Grade=="G_2018"&h1n1_pre_group=="Negative")
sub_titer1<- sub_titer[, c("pre_H1N1", "diff_H1N1")]
sub_titer1$group<- sub_titer1$diff_H1N1
sub_titer1$group[sub_titer1$group >5] = "high"
sub_titer1$group[sub_titer1$group <6] = "low"
sub_titer2<- subset(sub_titer1, group=="high")
student_id<- rownames(sub_titer2)
physeq.tib3<- subset_samples(physeq, type=="fecal"&national=="tibetan"& grade=="grade_18"&stage=="stage1")
physeq.tib<-  physeq.tib3
physeq.tib.p<-  prune_taxa(taxa_sums(physeq.tib) > 0, physeq.tib)
sample_data(physeq.tib.p)$human <- get_variable(physeq.tib.p, "student_id") %in% student_id
sub_physeq.tib.p<- subset_samples(physeq.tib.p, human=="TRUE")
physeq.fp<-  prune_taxa(taxa_sums(sub_physeq.tib.p) > 0, sub_physeq.tib.p)
high<- divergence(physeq.fp,apply(abundances(physeq.fp), 1, median), method = "bray")
sub_titer2$divergence <- unlist(high)
high.d <- sub_titer2

sub_titer2<- subset(sub_titer1, group=="low")
student_id<- rownames(sub_titer2)
physeq.tib3<- subset_samples(physeq, type=="fecal"&national=="tibetan"& grade=="grade_18"&stage=="stage1")
physeq.tib<-  physeq.tib3
physeq.tib.p<-  prune_taxa(taxa_sums(physeq.tib) > 0, physeq.tib)
sample_data(physeq.tib.p)$human <- get_variable(physeq.tib.p, "student_id") %in% student_id
sub_physeq.tib.p<- subset_samples(physeq.tib.p, human=="TRUE")
physeq.fp<-  prune_taxa(taxa_sums(sub_physeq.tib.p) > 0, sub_physeq.tib.p)
low<- divergence(physeq.fp,apply(abundances(physeq.fp), 1, median), method = "bray")
sub_titer2$divergence <- unlist(low)
low.d <- sub_titer2

data.all <- rbind(high.d, low.d)

p.d1<- ggplot(data.all, aes(x=group, y=divergence), shape=8 ) +
    theme_bw()+ 
    geom_boxplot(aes(x=group, fill=group), notch=FALSE, outlier.colour="black") +
    stat_summary(aes(x=group), fun.y=mean, geom="point", color="black") +
    scale_fill_manual(values=c("#E41A1C", "#377EB8"))+
    theme(axis.title.x =element_blank())+ #设置x.y标题上文本的名称
    theme(axis.text.x = element_text(size=14),axis.text.y = element_text(size=14)) + # 设置x.y坐标上文本大小
    theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank()) + #不显示网格线
    theme(legend.title = element_blank())+ # 不显示图例的标题
    theme(legend.position='none')+
    theme(title=element_text(size=18,color="#4F4F4F"))+
    theme(plot.title = element_text(hjust = 0.5))+
    labs(y="Divergence")+
    stat_compare_means(comparisons = my_comparisons, paired = FALSE,  color="red")

p.d1 # fig014

## 两个人群合并计算Divergence，高低应答组仍有显著差异
titer<- read.csv("C:/Users/hui/Desktop/TIV16S2020.lihui/processing.data/meta_immune_data.csv",header = T, sep = ',',row.names = 1)
sub_titer<- subset(titer, Grade=="G_2018"&h1n1_pre_group=="Negative")
sub_titer1<- sub_titer[, c("pre_H1N1", "diff_H1N1")]
sub_titer1$group<- sub_titer1$diff_H1N1
sub_titer1$group[sub_titer1$group >5] = "high"
sub_titer1$group[sub_titer1$group <6] = "low"
student_id<- rownames(sub_titer1)
physeq.tib3<- subset_samples(physeq, type=="fecal"&national=="tibetan"& grade=="grade_18"&stage=="stage1")
physeq.tib<-  physeq.tib3
physeq.tib.p<-  prune_taxa(taxa_sums(physeq.tib) > 0, physeq.tib)
sample_data(physeq.tib.p)$human <- get_variable(physeq.tib.p, "student_id") %in% student_id
sub_physeq.tib.p<- subset_samples(physeq.tib.p, human=="TRUE")
physeq.fp<-  prune_taxa(taxa_sums(sub_physeq.tib.p) > 0, sub_physeq.tib.p)
low<- divergence(physeq.fp,apply(abundances(physeq.fp), 1, median), method = "bray")
sub_titer1$divergence <- unlist(low)

p.d2 <- ggplot(sub_titer1, aes(x=group, y=divergence), shape=8 ) +
    theme_bw()+ 
    geom_boxplot(aes(x=group, fill=group), notch=FALSE, outlier.colour="black") +
    stat_summary(aes(x=group), fun.y=mean, geom="point", color="black") +
    scale_fill_manual(values=c("#E41A1C", "#377EB8"))+
    theme(axis.title.x =element_blank())+ #设置x.y标题上文本的名称
    theme(axis.text.x = element_text(size=14),axis.text.y = element_text(size=14)) + # 设置x.y坐标上文本大小
    theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank()) + #不显示网格线
    theme(legend.title = element_blank())+ # 不显示图例的标题
    theme(legend.position='none')+
    theme(title=element_text(size=18,color="#4F4F4F"))+
    theme(plot.title = element_text(hjust = 0.5))+
    labs(y="Divergence")+
    stat_compare_means(comparisons = my_comparisons, paired = FALSE,  color="black")

genus <- read.table('C:/Users/hui/Desktop/TIV16S2020.lihui/output.data/genus.fp.forcore.txt', header = T, sep = '\t', row.names = 1)
sub_titer1$Prevotella <- t(genus["g_Prevotella",])

p.p <- ggplot(data=sub_titer1, aes(x =Prevotella*100 , y = divergence)) +
    geom_point(aes(color=group),size=2)+ 
    scale_colour_manual(values=c("#E41A1C", "#377EB8"))+
    geom_vline(xintercept = 0, color = 'gray', size = 0.4) + 
    geom_smooth(method=lm, size=2, color="black")+ 
    labs(y= "Divergence", x="RA% of Prevotella")+
    theme(axis.text.x = element_text(size=12),axis.text.y = element_text(size=12)) +
    theme(legend.title = element_blank())+ 
    theme(title=element_text(size=14,color="#4F4F4F"))+
    theme(legend.position="none")+
    stat_cor(data=sub_titer1, method = "pearson", color="black", size=6)+
    theme_bw()+ 
    theme(axis.text.x = element_text(size=14),axis.text.y = element_text(size=14)) + 
    theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank()) + 
    theme(legend.position="none")+
    theme(title=element_text(size=14,color="#4F4F4F"))+
    theme(plot.title = element_text(hjust = 0.5))

(p.d2+ p.p+ plot_layout(nrow = 1, widths = c(1, 2)))# fig015

```

## 2.2 高应答人群的肠道菌群更加稳定性

```{r}
data.b<- read.delim("C:/Users/hui/Desktop/TIV16S2020.lihui/bray.tsv", sep = '\t', header = T, row.names = 1)
genus.all <- read.csv("C:/Users/hui/Desktop/TIV16S2020.lihui/all.genus.csv",header = T, sep = ',', row.names = 1)

meta <- meta(physeq)

data12<- as.data.frame(table(subset(meta, national=="tibetan" & type=="fecal" & stage %in% c("stage1", "stage2"))$student_id))
meta12 <- subset(meta, national=="tibetan" & type=="fecal" & stage %in% c("stage1", "stage2") & student_id %in% subset(data12, Freq==2)$Var1)
meta1<- rownames(subset(meta12, stage=="stage1"))
meta2<- rownames(subset(meta12, stage=="stage2"))
Bray_Curtis <- diag(as.matrix(data.b[meta1, meta2]))
bray.f1<- subset(meta12, stage=="stage1")
bray.f1$Bray_Curtis <- Bray_Curtis
bray.f1$Prevotella <- t(genus.all["g_Prevotella", meta2]-genus.all["g_Prevotella", meta1])

sub_titer1$Bray_Curtis <- subset(bray.f1, student_id %in% rownames(sub_titer1))$Bray_Curtis
sub_titer1$Prevotella <- subset(bray.f1, student_id %in% rownames(sub_titer1))$Prevotella

p3<- ggplot(sub_titer1, aes(x=group, y=Bray_Curtis), shape=8 ) +
    theme_bw()+ 
    geom_boxplot(aes(x=group, fill=group), notch=FALSE, outlier.colour="black") +
    stat_summary(aes(x=group), fun.y=mean, geom="point", color="black") +
    scale_fill_manual(values=c("#E41A1C", "#377EB8"))+
    theme(axis.title.x =element_blank())+ #设置x.y标题上文本的名称
    theme(axis.text.x = element_text(size=14),axis.text.y = element_text(size=14)) + # 设置x.y坐标上文本大小
    theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank()) + #不显示网格线
    theme(legend.title = element_blank())+ # 不显示图例的标题
    theme(legend.position='none')+
    theme(title=element_text(size=18,color="#4F4F4F"))+
    theme(plot.title = element_text(hjust = 0.5))+
    labs(y="Paired Bray-Curtis Disytance")+
    stat_compare_means(comparisons = my_comparisons, paired = FALSE,  color="black")

p4<- ggplot(data=sub_titer1, aes(x =Prevotella*100 , y = Bray_Curtis)) +
    geom_point(aes(color=group),size=2)+ 
    scale_colour_manual(values=c("#E41A1C", "#377EB8"))+ 
    geom_smooth(method=lm, size=2, color="black")+ 
    labs(y= "Paired Bray-Curtis Disytance", x="Changes of Prevotella")+
    theme(axis.text.x = element_text(size=12),axis.text.y = element_text(size=12)) +
    theme(legend.title = element_blank())+ 
    theme(title=element_text(size=14,color="#4F4F4F"))+
    theme(legend.position="none")+
    stat_cor(data=sub_titer1, method = "pearson", color="black", size=6)+
    theme_bw()+ 
    theme(axis.text.x = element_text(size=14),axis.text.y = element_text(size=14)) + 
    theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank()) + 
    theme(legend.position="none")+
    theme(title=element_text(size=14,color="#4F4F4F"))+
    theme(plot.title = element_text(hjust = 0.5))

(p3+ p4+ plot_layout(nrow = 1, widths = c(1, 2))) #fig016
```


## 2.3 高-低应答人群肠道菌群的网络差异（高应答人群体现PARS强互作）

```{r}

library(Matrix)
library(car)
library(igraph)
library(RCy3)
library(ggrepel)

load('C:/Users/hui/Desktop/TIV16S2020.lihui/processing.data/physeq.tiv16sall.Rdata')
  physeq<- physeq.tiv16sall
  titer<- read.csv("C:/Users/hui/Desktop/TIV16S2020.lihui/processing.data/meta_immune_data.csv",header = T, sep = ',',row.names = 1)
  sub_titer<- subset(titer, Grade=="G_2018"&h1n1_pre_group=="Negative")
  sub_titer1<- sub_titer[, c("pre_H1N1", "diff_H1N1")]
  sub_titer1$group<- sub_titer1$diff_H1N1
  sub_titer1$group[sub_titer1$group >5] = "high"
  sub_titer1$group[sub_titer1$group <6] = "low"
  student_id<- rownames(sub_titer1)
  physeq.tib3<- subset_samples(physeq, type=="fecal"&national=="tibetan"& grade=="grade_18"&stage=="stage1")
  physeq.tib<-  physeq.tib3
  physeq.tib.p<-  prune_taxa(taxa_sums(physeq.tib) > 0, physeq.tib)
  sample_data(physeq.tib.p)$human <- get_variable(physeq.tib.p, "student_id") %in% student_id
  sub_physeq.tib.p<- subset_samples(physeq.tib.p, human=="TRUE")
  physeq.fp<-  prune_taxa(taxa_sums(sub_physeq.tib.p) > 0, sub_physeq.tib.p)

otu.f<- read.table('C:/Users/hui/Desktop/TIV16S2020.lihui/output.data/genus.fp.forcore.txt', header = T, sep = '\t', row.names = 1)
core.taxa<- core_members(otu.f, detection = 0.0002, prevalence = 3/26)
core.taxa.f<- core.taxa[-c(8, 53, 59, 77, 89, 90)]
otu.low<- 10000*t(otu.f[core.taxa.f, rownames(subset(sub_titer1, group=="low"))])
otu.high<- 10000*t(otu.f[core.taxa.f, rownames(subset(sub_titer1, group=="high"))])

betaMat.low = read.table('C:/Users/hui/Desktop/TIV16S2020.lihui/output.data/sparcc.low.high/low.cor.txt', header = T, sep = '\t', row.names = 1)
betaMat.high = read.table('C:/Users/hui/Desktop/TIV16S2020.lihui/output.data/sparcc.low.high/high.cor.txt', header = T, sep = '\t', row.names = 1)
p.high<- read.table('C:/Users/hui/Desktop/TIV16S2020.lihui/output.data/sparcc.low.high/high.pvals.txt', header = T, sep = '\t', row.names = 1)
p.low<- read.table('C:/Users/hui/Desktop/TIV16S2020.lihui/output.data/sparcc.low.high/low.pvals.txt', header = T, sep = '\t', row.names = 1)

tax<- as.data.frame(physeq.fp@tax_table@.Data)
tax1<- subset(tax, Genus %in% c(core.taxa.f))
tax2<- dplyr::filter(tax1, !duplicated(tax1$Genus))
rownames(tax2)<- tax2$Genus
tax3<- tax2[colnames(otu.high),]

betaMat.high[p.high>0.05] = 0
betaMat.low[p.low>0.05] = 0
betaMat.high<- as.matrix(betaMat.high)
betaMat.low<- as.matrix(betaMat.low)

high <- betaMat.high

low<- betaMat.low

otu.low100<- t(otu.low)/100
otu.high100<- t(otu.high)/100
otu.low100<- as.data.frame(otu.low100)[rownames(low),]
otu.high100<- as.data.frame(otu.high100)[rownames(high),]
otu.low100$rel<- rowMeans(otu.low100)
otu.high100$rel<- rowMeans(otu.high100)
igraph.size.high <- recode(otu.high100$rel,"0:0.001=2;0.001:0.1=4;0.1:1=6;1:5=10;5:15=12;15:100=15")
igraph.size.low <- recode(otu.low100$rel,"0:0.001=2;0.001:0.1=4;0.1:1=6;1:5=10;5:15=12;15:100=15")


tax.high<- tax3[rownames(high),]
tax.low<- tax3[rownames(low),]

# 高应答组肠道菌群共线网络
igraph.high <-  graph_from_adjacency_matrix(high,
                                     mode="undirected",
                                     weighted=TRUE,
                                     diag=FALSE )

igraph.weight = E(igraph.high)$weight
E(igraph.high)$weight = NA
sum(igraph.weight>0)# number of postive correlation
sum(igraph.weight<0)# number of negative correlation
E.color = igraph.weight
E.color = ifelse(E.color>0, "red",ifelse(E.color<0, "blue","grey"))
E(igraph.high)$color = as.character(E.color)
E(igraph.high)$width = recode(abs(igraph.weight),"0.6:0.7=2;0.7:0.8=4;0.8:1.0=6")
                                    
igraph.col.high <- as.character(recode(tax.high$Phylum,"'p_Bacteroidetes'='#7fbc41';'p_Firmicutes'='black';'p_Fusobacteria'='#c51b7d';'p_Proteobacteria'='#6a3d9a';'p_Lentisphaerae'='#cab2d6';'p_Verrucomicrobia'='#ff7f00'; 'p_Actinobacteria'='yellow'; 'p_Tenericutes'='grey'; 'p_Euryarchaeota'='red'")) 

igraph.col.low <- as.character(recode(tax.low$Phylum,"'p_Bacteroidetes'='#7fbc41';'p_Firmicutes'='black';'p_Fusobacteria'='#c51b7d';'p_Proteobacteria'='#6a3d9a';'p_Lentisphaerae'='#cab2d6';'p_Verrucomicrobia'='#ff7f00'; 'p_Actinobacteria'='yellow'; 'p_Tenericutes'='grey'; 'p_Euryarchaeota'='red'"))

V(igraph.high)$phylum =  as.character(recode(tax.high$Phylum,"'p_Bacteroidetes'='#7fbc41';'p_Firmicutes'='black';'p_Fusobacteria'='#c51b7d';'p_Proteobacteria'='#6a3d9a';'p_Lentisphaerae'='#cab2d6';'p_Verrucomicrobia'='#ff7f00'; 'p_Actinobacteria'='yellow'; 'p_Tenericutes'='grey'; 'p_Euryarchaeota'='red'"))

V(igraph.high)$size=  recode(otu.high100$rel,"0:0.001=2;0.001:0.1=4;0.1:1=6;1:5=10;5:15=12;15:100=15")

# createNetworkFromIgraph(igraph.high,"cy.high")

# 低应答组肠道菌群共线网络
igraph.low <-  graph_from_adjacency_matrix(low,
                                     mode="undirected",
                                     weighted=TRUE,
                                     diag=FALSE )

igraph.weight = E(igraph.low)$weight
E(igraph.low)$weight = NA
sum(igraph.weight>0)# number of postive correlation
sum(igraph.weight<0)# number of negative correlation
E.color = igraph.weight
E.color = ifelse(E.color>0, "red",ifelse(E.color<0, "blue","grey"))
E(igraph.low)$color = as.character(E.color)
E(igraph.low)$width = recode(abs(igraph.weight),"0.6:0.7=2;0.7:0.8=4;0.8:1.0=6")
                                    
igraph.col.low <- as.character(recode(tax.low$Phylum,"'p_Bacteroidetes'='#7fbc41';'p_Firmicutes'='black';'p_Fusobacteria'='#c51b7d';'p_Proteobacteria'='#6a3d9a';'p_Lentisphaerae'='#cab2d6';'p_Verrucomicrobia'='#ff7f00'; 'p_Actinobacteria'='yellow'; 'p_Tenericutes'='grey'; 'p_Euryarchaeota'='red'")) 

igraph.col.low <- as.character(recode(tax.low$Phylum,"'p_Bacteroidetes'='#7fbc41';'p_Firmicutes'='black';'p_Fusobacteria'='#c51b7d';'p_Proteobacteria'='#6a3d9a';'p_Lentisphaerae'='#cab2d6';'p_Verrucomicrobia'='#ff7f00'; 'p_Actinobacteria'='yellow'; 'p_Tenericutes'='grey'; 'p_Euryarchaeota'='red'"))

V(igraph.low)$phylum =  as.character(recode(tax.low$Phylum,"'p_Bacteroidetes'='#7fbc41';'p_Firmicutes'='black';'p_Fusobacteria'='#c51b7d';'p_Proteobacteria'='#6a3d9a';'p_Lentisphaerae'='#cab2d6';'p_Verrucomicrobia'='#ff7f00'; 'p_Actinobacteria'='yellow'; 'p_Tenericutes'='grey'; 'p_Euryarchaeota'='red'"))

V(igraph.low)$size=  recode(otu.low100$rel,"0:0.001=2;0.001:0.1=4;0.1:1=6;1:5=10;5:15=12;15:100=15")

# sudo bash Desktop/常用软件/cytoscape-unix-3.8.0/cytoscape.sh 
# createNetworkFromIgraph(igraph.low,"cy.low")

#网络图生成为lowvshigh.cys


# 使用热图展示与Prevotella关联的互作物种
betaMat.low = read.table('C:/Users/hui/Desktop/TIV16S2020.lihui/output.data/sparcc.low.high/low.cor.txt', header = T, sep = '\t', row.names = 1)
betaMat.high = read.table('C:/Users/hui/Desktop/TIV16S2020.lihui/output.data/sparcc.low.high/high.cor.txt', header = T, sep = '\t', row.names = 1)
p.high<- read.table('C:/Users/hui/Desktop/TIV16S2020.lihui/output.data/sparcc.low.high/high.pvals.txt', header = T, sep = '\t', row.names = 1)
p.low<- read.table('C:/Users/hui/Desktop/TIV16S2020.lihui/output.data/sparcc.low.high/low.pvals.txt', header = T, sep = '\t', row.names = 1)


colnames(betaMat.high) <- rownames(betaMat.high)
colnames(betaMat.low) <- rownames(betaMat.low)
colnames(p.high) <- rownames(p.high)
colnames(p.low) <- rownames(p.low)

betaMat.high[p.high>0.05] = 0
betaMat.low[p.low>0.05] = 0
betaMat.high<- as.data.frame(betaMat.high)
betaMat.low<- as.data.frame(betaMat.low)

high.rnames <- rownames(betaMat.high[which(betaMat.high$g_Prevotella !=0), which(betaMat.high$g_Prevotella !=0)])
high.cnames <- colnames(betaMat.high[which(betaMat.high$g_Prevotella !=0), which(betaMat.high$g_Prevotella !=0)])

low.rnames <- rownames(betaMat.low[which(betaMat.low$g_Prevotella !=0), which(betaMat.low$g_Prevotella !=0)])
low.cnames <- colnames(betaMat.low[which(betaMat.low$g_Prevotella !=0), which(betaMat.low$g_Prevotella !=0)])


 rnames <- union(high.rnames, low.rnames)
 rnames1 <- c(rnames, "g_Prevotella")

genus.p = tax_glom(physeq.fp, "Genus")
data<- genus.p@tax_table@.Data
data1<- as.data.frame(data)
data2<- data1[!duplicated(data1$Genus), ]
c1<- rownames(subset(data2, Genus %in% rnames1))
ex1 = prune_taxa(c1, genus.p)
library(ggtree)
p.tree<- ggtree(ex1, aes(color=Phylum), size=1 )+ 
    geom_tiplab(aes(label=Genus),align=T, linesize=.5)+ 
    geom_tippoint(aes(color=Phylum), size=2)+ coord_cartesian(clip = 'off')+
    theme_tree2(plot.margin=margin(0, 400, 6, 6))+  
    theme(legend.position = c(0.15, 0.8))

data1<- na.omit(p.tree$data[!duplicated(p.tree$data$y),])
ord<- data1[order(data1$y),]$Genus

genus <- read.table('C:/Users/hui/Desktop/TIV16S2020.lihui/output.data/genus.fp.forcore.txt', header = T, sep = '\t', row.names = 1)

sub_design<- sub_titer1
sampFile = as.data.frame(sub_design$group,row.names = row.names(sub_design))
colnames(sampFile)[1] = "group"
mat_t = t(genus[as.character(ord),])
mat_t2 = merge(sampFile, mat_t, by="row.names")[,-1]
mat_mean = aggregate(mat_t2[,-1], by=mat_t2[1], FUN=mean) # mean
mat_mean_final = do.call(rbind, mat_mean)[-1,]
colnames(mat_mean_final) = mat_mean$group
mat <- mat_mean_final
rownames(mat) <- data1[order(data1$y),]$label

p.tree1<- ggtree::gheatmap(p.tree, mat, low="black", high="white", width=0.25,offset = 0.6, font.size=4,colnames_angle=45,  colnames_position = "top") +  theme(legend.position="none")

 # fig017


high.genus <- betaMat.high[as.character(ord), as.character(ord)]
 low.genus <- betaMat.low[as.character(ord), as.character(ord)]
 
 rownames(high.genus) <- LETTERS[1:22]
 rownames(low.genus) <- LETTERS[1:22]
 colnames(high.genus) <- LETTERS[1:22]
 colnames(low.genus) <- LETTERS[1:22]
 
library(vctrs)
library(grid)

geom_rectriangle <- function(mapping = NULL, data = NULL,
                             stat = "identity", position = "identity",
                             ...,
                             linejoin = "mitre",
                             na.rm = FALSE,
                             show.legend = NA,
                             inherit.aes = TRUE) {
    layer(
        data = data,
        mapping = mapping,
        stat = stat,
        geom = GeomRectriangle,
        position = position,
        show.legend = show.legend,
        inherit.aes = inherit.aes,
        params = list(
            linejoin = linejoin,
            na.rm = na.rm,
            ...
        )
    )
}

GeomRectriangle <- ggproto(
    "GeomRectriangle", Geom,
    default_aes = aes(r = 1, colour = "#e0e0e0", fill = NA, size = 0.25, linetype = 1,
                      alpha = NA,type = "upper"),
    required_aes = c("x", "y"),
    draw_panel = function(self, data, panel_params, coord, linejoin = "mitre",type = "upper") {
        aesthetics <- setdiff(names(data), c("x", "y"))
        
        polys <- lapply(split(data, seq_len(nrow(data))), function(row) {
            rectriangle <- point_to_rectriangle(row$x, row$y, row$r, row$type)
            aes <- new_data_frame(row[aesthetics])[rep(1, 4), ]
            GeomPolygon$draw_panel(cbind(rectriangle, aes), panel_params, coord)
        })
        
        ggplot2:::ggname("geom_rectriangle", do.call("grobTree", polys))
    },
    draw_key = draw_key_polygon
)


point_to_rectriangle <- function(x, y, r, type = type) {
    r <- 0.5 * sign(r) * sqrt(abs(r))
    #r0 = 0.5
    xmin <- - r + x
    xmax <- r + x
    ymin <- - r + y
    ymax <- r + y 
    if(type == "upper"){
        df = new_data_frame(list(
            y = c(ymax, ymax, ymin, ymax),
            x = c(xmin, xmax, xmin, xmin)
        ))
    }else if(type == "lower"){
        df = new_data_frame(list(
            y = c(ymax, ymin, ymin, ymax),
            x = c(xmax, xmax, xmin, xmax) 
        ))
    }
    df
}

library(tibble)
library(tidyr)

data <- cor(high) %>%
    as.data.frame()%>% 
    rownames_to_column("A") %>% 
    gather("B","value1",-A) %>% 
    mutate(value2 = -log2(sample(value1, length(value1))+1))


#install.packages("ggnewscale")
get_lower_tri<-function(cormat){
    cormat[upper.tri(cormat)] <- NA
    return(cormat)
}
#####获取相关性矩阵的下三角矩阵
get_upper_tri <- function(cormat){
    cormat[lower.tri(cormat)]<- NA
    return(cormat)
}
high1 <- get_upper_tri(high.genus)
low1 <- get_upper_tri(low.genus)

 data.low <- low1 %>%
    as.data.frame()%>% 
    rownames_to_column("A1") %>% 
    gather("B1","low",-A1)
 
 data.high <- high1 %>%
    as.data.frame()%>% 
    rownames_to_column("A1") %>% 
    gather("B1","high",-A1)

low <- data.low$low
data<- cbind(data.high, low)

p<- ggplot()+
    geom_rectriangle(data = data, aes(A1, B1, fill = high),type = "upper", r = 1)+
    scale_fill_gradient2(high = "#762a83",low = "#1b7837", na.value = "#e0e0e0")+
    ggnewscale::new_scale_fill()+
    geom_rectriangle(data = data, aes(A1, B1, fill = low),type = "lower", r = 1)+
    scale_fill_gradient2(high = "#762a83",low = "#1b7837", na.value = "#e0e0e0")+
    labs(x = "", y = "")+
    scale_x_discrete(position = "top", labels= as.character(ord))+
    scale_y_discrete(labels = NULL)+
    theme_bw()

p.cor<- p+ theme(axis.text.x = element_text(hjust=0, vjust=0, angle = 45))

p.cor #fig018

# 强弱应答人群的Closeness Centrality和Betweenness Centrality差异
net<- read.csv("C:/Users/hui/Desktop/TIV16S2020.lihui/output.data/sparcc.low.high/low.csv",header = T, sep = ',')

net1 <- subset(net, groups %in% c(1,2))
find_hull <- function(df) df[chull(df[[1]], df[[2]]), ]
hulls <- ddply(net1, "groups", find_hull)
ggplot(net, aes(x=ClosenessCentrality, y=BetweennessCentrality, shape=group)) +
    theme(panel.grid = element_line(color = 'gray', linetype = 2, size = 0.1), panel.background = element_rect(color = 'black', fill = 'transparent'), legend.key = element_rect(fill = 'transparent')) + #去掉背景框
    geom_point(aes(color = name), alpha=0.8, size = 3) + #可在这里修改点的透明度、大小
    scale_color_manual(values=c('#1b9e77', '#d95f02','#7570b3','#e7298a',"grey")) + #可在这里修改点的颜色
    guides(fill = guide_legend(order = 1), shape = guide_legend(order = 2), color = guide_legend(order = 3))+  #设置图例展示顺序
    labs(x = "Closeness Centrality", y = "Betweenness Centrality")+
    theme(axis.text.x = element_text(size=14),axis.text.y = element_text(size=14)) + # 设置x.y坐标上文本大小
    theme(title=element_text(size=14,color="#4F4F4F"))+
    theme(plot.title = element_text(hjust = 0.5))+ 
    theme(legend.title = element_blank(),legend.text = element_text(colour="#4F4F4F", size = 14))+
    theme(strip.text = element_text(colour = 'black', face = 'bold', size = rel(1.2)), strip.background = element_rect(fill = 'white', colour = 'black', size = rel(2), linetype = 1))+
    geom_polygon(data = hulls, alpha = 0.5, aes(fill = factor(groups)),show.legend = F)+
    scale_fill_manual(values=c("#E41A1C", "#377EB8"))
```

# 3. Prevotella影响TIV应答的相关机制

```{r}
#宏基因组-免疫-代谢多组学关联
#基于宏基因组的LPS、Flagelin、CpG-DNA预测
#血清、粪便等体外实验检测相关分子佐证
```

```{r}

library(mixOmics)
load('C:/Users/hui/Desktop/mixomics/mixomcs4.Rdata')

X <- mixomics4
Y= as.factor(sub_titer1$group)

MyResult.diablo <- block.splsda(X, Y)
plotDiablo(MyResult.diablo, ncomp = 1)

list.keepX <- list(species = 5, bolite = 5, cytokine = 5, pathway = 5)
MyResult.diablo <- block.splsda(X, Y, keepX=list.keepX)

Myauc.diablo <- auroc(MyResult.diablo, roc.block = c("species", "pathway", "bolite", "cytokine")  , roc.comp = 1)

graph.species <- Myauc.diablo[["graph.species"]][["comp1"]][["data"]]
graph.pathway <- Myauc.diablo[["graph.pathway"]][["comp1"]][["data"]]
graph.bolite <- Myauc.diablo[["graph.bolite"]][["comp1"]][["data"]]
graph.cytokine <- Myauc.diablo[["graph.cytokine"]][["comp1"]][["data"]]
graph.species$omics <- "species: AUC=0.875" 
graph.pathway$omics <- "pathway: AUC=0.844"
graph.bolite$omics <- "bolite: AUC=0.806"
graph.cytokine$omics <- "cytokine: AUC=0.675"
plotdata <- rbind(graph.cytokine, graph.bolite, graph.pathway, graph.species)
plotdata$omics <- factor(plotdata$omics, levels = c("species: AUC=0.875","pathway: AUC=0.844", "bolite: AUC=0.806", "cytokine: AUC=0.675"), ordered=TRUE)
ggplot(plotdata) + 
    geom_line(aes(x = Specificity, y = Sensitivity, color = omics), size=2) + 
    geom_abline(a=0, b=1, col="grey",lwd=1,lty=2)+
    scale_colour_manual(values=c("#9932cc", "#999900", "#90ee90", "#009999")) +
    theme(plot.title = element_text(face = 'bold',size=15))+
    theme_bw()+ 
    theme(axis.text.x = element_text(size=14),axis.text.y = element_text(size=14), legend.text=element_text(size=12)) + 
    theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank()) + 
    theme(legend.title = element_blank())+ 
    theme(legend.position='none')+
    theme(title=element_text(size=14,color="#4F4F4F"))+
    labs(x = "Specificity(%)", y = "Sensitivity(%)")+
    theme(legend.position = c(0.7, 0.25))

filter <- read.csv("C:/Users/hui/Desktop//mixomics/all.filter.csv",header = T, sep = ',')

filter <- read.csv("C:/Users/hui/Desktop/TIV-Manuscript & Figure/Figure/Fig2/mixomics/all.filter.csv",header = T, sep = ',', row.names = 1)

```
