---
title: "Prevalence of gut Prevotella copri significantly correlated with TIV-specific antibody production"
output: 
html_document:
      toc: yes
      theme: cerulean
---

          
```{r setup, include=FALSE}
	knitr::opts_chunk$set(echo = TRUE, warning = FALSE, tidy=TRUE, collapse=TRUE)
```


# figure 1b-c

```{r}
library(ggplot2)
library(plyr)
library(readr)
library(reshape2)
library(patchwork)
library(cowplot)
library(ggpubr) 
	
	
titer<- read.csv("Source Data/meta_immune_data.csv",header = T, sep = ',', row.names = 1)
	

titer$records <- titer$Grade
titer$records[titer$Grade %in% c("G1", "G2")] = "Record"
titer$records[titer$Grade == "G3"] = "no Record"

titer$records <- factor(titer$records, levels = c( "Record","no Record"), ordered=TRUE)

my_comparisons <- list(c("Record", "no Record"))

p.fig1b<- ggplot(titer, aes(x=records, y=diff_H1N1), shape=8 ) +
	    geom_boxplot(aes(x=records, fill=records), notch=FALSE) +
	    stat_summary(aes(x=records), fun.y=mean, geom="point") +
	    geom_jitter(width=0.05, height=0, size=2) +
	    scale_fill_manual(values=c("#b34e7e", "#1b7837"))+
	    theme(axis.title.x =element_blank())+
      theme_bw()+
	    theme(axis.text.x = element_text(size=14),axis.text.y = element_text(size=14)) + 
	    theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank()) + 
	    theme(legend.title = element_blank())+ 
	    theme(legend.position='none')+
	    theme(title=element_text(size=18,color="#4F4F4F"))+
	    scale_y_continuous(limits=c(0, 11), breaks=seq(0, 10, 2))+
	    theme(plot.title = element_text(hjust = 0.5))+
	    stat_compare_means(comparisons = my_comparisons, paired = FALSE, label = "p.signif", color="black", size=6)+
	    labs(y="H1N1 titer change (log)", x="")

my_comparisons <- list(c("Negative", "Positive"))
p.fig1c<- ggplot(sub_data, aes(x=h1n1_pre_group, y=diff_H1N1), shape=8 ) +
	    geom_boxplot(aes(x=h1n1_pre_group, fill=h1n1_pre_group), notch=FALSE) +
	    stat_summary(aes(x=h1n1_pre_group), fun.y=mean, geom="point") +
	    geom_jitter(width=0.05, height=0, size=2) +
	    scale_fill_manual(values=c("#533633", "#149684"))+
	    theme(axis.title.x =element_blank())+
      theme_bw()+
	    theme(axis.text.x = element_text(size=14),axis.text.y = element_text(size=14)) + 
	    theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank()) + 
	    theme(legend.title = element_blank())+ 
	    theme(legend.position='none')+
	    theme(title=element_text(size=18,color="#4F4F4F"))+
	    scale_y_continuous(limits=c(0, 11), breaks=seq(0, 10, 2))+
	    theme(plot.title = element_text(hjust = 0.5))+
	    stat_compare_means(comparisons = my_comparisons, paired = FALSE, label = "p.signif", color="black", size=6)+
	    labs(y="", x="")
	    
p.fig1b | p.fig1c 

```

# figure 1e-f 

```{r}
# These LEfSe result were performed at http://huttenhower.sph.harvard.edu/galaxy/
# 16S rRNA gene squencing data file was used: Figure 1/genus.fecal.lowhigh.txt, LDA cut off > 2.5
# metagenomics squencing data file was used: Figure 1/species.fecal.lowhigh.txt, LDA cut off > 2.5

```

# figure 1d,g

```{r}
library(microbiome)
library(phyloseq)
library(vegan)

# G1 = G_2016
# G2 = G_2017
# G3 = G_2018
# G4 = G_2019

load('Source Data/physeq.tiv16sall.Rdata') # Created by phyloseq package
physeq<- physeq.tiv16sall
titer<- read.csv("Source Data/meta_immune_data.csv",header = T, sep = ',',row.names = 1)
sub_titer<- subset(titer, Grade=="G_2018"&h1n1_pre_group=="Negative")
sub_titer1<- sub_titer[, c("pre_H1N1", "diff_H1N1")]
sub_titer1$group<- sub_titer1$diff_H1N1
sub_titer1$group[sub_titer1$group >5] = "high"
sub_titer1$group[sub_titer1$group <6] = "low"
student_id<- rownames(sub_titer1)
physeq.tib3<- subset_samples(physeq, type=="fecal"&national=="tibetan"& grade=="grade_18"&stage=="stage1")
physeq.tib<-  physeq.tib3
physeq.tib.p<-  prune_taxa(taxa_sums(physeq.tib) > 0, physeq.tib)
sample_data(physeq.tib.p)$human <- get_variable(physeq.tib.p, "student_id") %in% student_id
sub_physeq.tib.p<- subset_samples(physeq.tib.p, human=="TRUE")
physeq.fp<-  prune_taxa(taxa_sums(sub_physeq.tib.p) > 0, sub_physeq.tib.p)
physeq.fp1<- filter_taxa(physeq.fp, function(x) sum(x > 3) > 1, TRUE)

dis_bray.f<- phyloseq::distance(physeq.fp1, "bray")
set.seed(002)
adon.results<-adonis(dis_bray.f~ sub_titer1$group, perm=999)
knitr::kable(adon.results[["aov.tab"]])


otu<- t(physeq.fp1@otu_table@.Data)
nmds <- metaMDS(otu, distance = 'bray', k = 2)

data.all<- cbind(nmds[["points"]], genus.core, sub_titer1)
nmds_plot.fecal <- ggscatter(data.all, x= "MDS1", y = "MDS2", 
                             color = "group", 
                             size = 3,
                             palette = "Set1", 
                             ellipse = TRUE, 
                             mean.point = FALSE, 
                             star.plot = TRUE, 
                             ellipse.level = FALSE,  
                             ggtheme = theme_minimal()) +
    labs(x = "NMDS1", y = "NMDS2") +
    ggtitle("NMDS")+
    theme(panel.grid = element_line(color = 'gray', linetype = 2, size = 0.1), panel.background = element_rect(color = 'black', fill = 'transparent'), legend.key = element_rect(fill = 'transparent')) + 
    geom_vline(xintercept = 0, color = 'gray', size = 0.4) + 
    geom_hline(yintercept = 0, color = 'gray', size = 0.4) + 
    theme(axis.text.x = element_text(size=16),axis.text.y = element_text(size=16)) + 
    labs(title="Bray_Curtis distance of Fecal samples")+
    theme(title=element_text(size=16,color="#4F4F4F"))+
    theme(plot.title = element_text(hjust = 0.5))+ 
    theme(legend.title = element_blank(),legend.text = element_text(colour="#4F4F4F", size = 14))+
    annotate("text",x=-0.35,y=0.4,parse=TRUE,size=4,label=paste('Adonis-P:',adon.results$aov.tab$`Pr(>F)`[1]),family="serif",fontface="italic",colour="black")+
    annotate("text",x=-0.35,y=0.35,parse=TRUE,size=4,label=paste("Stess: ", nmds[["stress"]]),family="serif",fontface="italic",colour="black")+
    theme(legend.position="none")

nmds_plot.fecal

data.all <-  read.csv("Figure 1/genus_speicies.csv",header = T, sep = ',',row.names = 1)

my_comparisons <- list(c("low", "high"))
p.Prevotella<- ggplot(data.all, aes(x=group, y=g_Prevotella*100), shape=8 ) +
    theme_bw()+ 
    geom_boxplot(aes(x=group, fill=group), notch=FALSE, outlier.colour="grey") +
    stat_summary(aes(x=group), fun.y=mean, geom="point", color="orange") +
    geom_jitter(width=0.2, size=3) +
    scale_fill_manual(values=c("#E41A1C", "#377EB8"))+
    theme(axis.title.x =element_blank())+ 
    theme(axis.text.x = element_text(size=14),axis.text.y = element_text(size=14)) +
    theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank()) +
    theme(legend.title = element_blank())+ 
    theme(legend.position='none')+
    theme(title=element_text(size=18,color="#4F4F4F"))+
    theme(plot.title = element_text(hjust = 0.5))+
    labs(y="Relative Aboundance %")+
    stat_compare_means(comparisons = my_comparisons, paired = FALSE,  color="red")+
    labs(title="Prevotella")

p.Prevotella 

my_comparisons <- list(c("low", "high"))
p.copri<- ggplot(data.all, aes(x=group, y=Prevotella_copri*100), shape=8 ) +
    theme_bw()+ 
    geom_boxplot(aes(x=group, fill=group), notch=FALSE, outlier.colour="grey") +
    stat_summary(aes(x=group), fun.y=mean, geom="point", color="orange") +
    geom_jitter(width=0.2, size=3) +
    scale_fill_manual(values=c("#E41A1C", "#377EB8"))+
    theme(axis.title.x =element_blank())+ 
    theme(axis.text.x = element_text(size=14),axis.text.y = element_text(size=14)) +
    theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank()) + 
    theme(legend.title = element_blank())+ 
    theme(legend.position='none')+
    theme(title=element_text(size=18,color="#4F4F4F"))+
    theme(plot.title = element_text(hjust = 0.5))+
    labs(y="Relative Aboundance %")+
    stat_compare_means(comparisons = my_comparisons, paired = FALSE,  color="red")+
    labs(title="Prevotella copri")

p.copri

```


